ReactWindows/ReactNative.Net46/Views/TextInput/PlaceholderAdorner.cs

```diff
@@ -0,0 +1,1104 @@
+        /// </summary>
+        public static readonly DependencyProperty FontWeightProperty =
+            DependencyProperty.RegisterAttached("FontWeight", typeof(FontWeight?), typeof(PlaceholderAdorner),
+                new FrameworkPropertyMetadata(null));
+
+        #endregion
+
+        #region FontStretch Property
+
+        /// <summary>
+        ///   Gets the value of the <see cref="T:ReactNative.Views.TextInput.PlaceholderAdorner" />.FontStretch attached property for a specified element.
+        /// </summary>
+        /// <param name="adornedElement">
+        ///   The element from which the property value is read.
+        /// </param>
+        /// <returns>
+        ///   The placeholder text font stretch property value for the element.
+        /// </returns>
+        /// <exception cref="T:ArgumentNullException">
+        ///   Raised when adornedElement is null.
+        /// </exception>
+        public static FontStretch? GetFontStretch(Control adornedElement)
+        {
+            if (adornedElement == null)
+            {
+                throw new ArgumentNullException(nameof(adornedElement));
+            }
+
+            return (FontStretch?)adornedElement.GetValue(FontStretchProperty);
+        }
+
+        /// <summary>
+        ///   Sets the value of the <see cref="T:ReactNative.Views.TextInput.PlaceholderAdorner" />.FontStretch attached property to a specified element.
+        /// </summary>
+        /// <param name="adornedElement">
+        ///   The element to which the attached property is written.
+        /// </param>
+        /// <param name="fontStretch">
+        ///   The placeholder text Font stretch value.
+        /// </param>
+        /// <exception cref="T:ArgumentNullException">
+        ///   Raised when adornedElement is null.
+        /// </exception>
+        /// <exception cref="T:InvalidOperationException">
+        ///   Raised when adornedElement is not a <see cref="T:System.Windows.Controls.TextBox"/>,
+        ///   <see cref="T:System.Windows.Controls.RichTextBox"/> or <see cref="T:System.Windows.Controls.PasswordBox"/>.
+        /// </exception>
+        public static void SetFontStretch(Control adornedElement, FontStretch? fontStretch)
+        {
+            if (adornedElement == null)
+            {
+                throw new ArgumentNullException(nameof(adornedElement));
+            }
+
+            adornedElement.SetValue(FontStretchProperty, fontStretch);
+        }
+
+        /// <summary>
+        ///   Identifies the <see cref="T:T:ReactNative.Views.TextInput.PlaceholderAdorner" />.FontStretch attached property.
+        /// </summary>
+        public static readonly DependencyProperty FontStretchProperty =
+            DependencyProperty.RegisterAttached("FontStretch", typeof(FontStretch?), typeof(PlaceholderAdorner), new PropertyMetadata(null));
+
+        #endregion
+
+        #region HideOnFocus Property
```

Comment:
Setting this boolean to true will change when the placeholder text is removed... By default this value is false, which means the placeholder text will remain until the first character is entered in the textbox... If this value is true then the placeholder text is removed when the text box receives focus and will return when it loses focus and the text box is empty... 

This pattern matches the behavior of UWP's placeholder text implemetnation.